libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java

```diff
@@ -72,72 +104,137 @@ public CiMitService(
         try {
-            payload =
-                    OBJECT_MAPPER.writeValueAsString(
-                            new PutCiRequest(govukSigninJourneyId, ipAddress, vc.getVcString()));
-
+            if (configService.enabled(CIMIT_API_GATEWAY_ENABLED)) {
+                var payload =
+                        OBJECT_MAPPER.writeValueAsString(
+                                new PostCiPrivateApiRequest(vc.getVcString()));
+
+                var uri = getUriBuilderWithBaseApiUrl(POST_CI_ENDPOINT).build();
+
+                var response = sendPostHttpRequest(uri, govukSigninJourneyId, ipAddress, payload);
+                var parsedResponse =
+                        OBJECT_MAPPER.readValue(response.body(), PrivateApiResponse.class);
+
+                if (FAILED_RESPONSE.equals(parsedResponse.result())) {
+                    logApiRequestError(parsedResponse);
+                    throw new CiPutException(FAILED_API_REQUEST);
+                }
+
+            } else {
+                var payload =
+                        OBJECT_MAPPER.writeValueAsString(
+                                new PutCiRequest(
+                                        govukSigninJourneyId, ipAddress, vc.getVcString()));
+
+                var invokeRequest =
+                        InvokeRequest.builder()
+                                .functionName(
+                                        configService.getEnvironmentVariable(
+                                                CI_STORAGE_PUT_LAMBDA_ARN))
+                                .payload(SdkBytes.fromUtf8String(payload))
+                                .qualifier(LIVE_ALIAS)
+                                .build();
+
+                var response = lambdaClient.invoke(invokeRequest);
+
+                if (lambdaExecutionFailed(response)) {
+                    logLambdaExecutionError(response, CI_STORAGE_PUT_LAMBDA_ARN);
+                    throw new CiPutException(FAILED_LAMBDA_MESSAGE);
+                }
+            }
         } catch (JsonProcessingException e) {
-            throw new CiPutException("Failed to serialize PutCiRequest");
-        }
-
-        var invokeRequest =
-                InvokeRequest.builder()
-                        .functionName(
-                                configService.getEnvironmentVariable(CI_STORAGE_PUT_LAMBDA_ARN))
-                        .payload(SdkBytes.fromUtf8String(payload))
-                        .qualifier(LIVE_ALIAS)
-                        .build();
-
-        LOGGER.info(LogHelper.buildLogMessage("Sending VC to CIMIT."));
-        var response = lambdaClient.invoke(invokeRequest);
-
-        if (lambdaExecutionFailed(response)) {
-            logLambdaExecutionError(response, CI_STORAGE_PUT_LAMBDA_ARN);
+            throw new CiPutException("Failed to serialize payload for post CI request.");
+        } catch (CiPutException e) {
+            throw e;
+        } catch (Exception e) {
```

Comment:
We very rarely want to catch blanket `Exception`s - can we be more specific about which exceptions you expect?